### ISSUE TITLE
Enable undocumented-param for pep257 docstrings

### ISSUE THREAD

**Issue by joelberkeley** [astral-sh/ruff#6887]
I'd like to be able to check for undocumented parameters whilst using pep257 docstrings. I'm specifically using reST style

**Comment by zanieb** [comment]
Hey @joelberkeley could you please share an example and check if there's support for this in pydocstyle?

[D417 docs](https://beta.ruff.rs/docs/rules/undocumented-param/)

**Comment by joelberkeley** [comment]
I'd like this to be flagged
```
def max(x: int, y: int) -> int:
    """
    :param x: An integer.
    :return: The largest of `x` and `y`.    
    """
    return x if x > y else y
```
It would also be good to flag a missing return
```
def max(x: int, y: int) -> int:
    """
    :param x: An integer.
    :param y: An integer.
    """
    return x if x > y else y
```
Is this D417 in pydocstyle?

**Comment by joelberkeley** [comment]
pydocstyle mentions D417 if that's what you mean http://www.pydocstyle.org/en/stable/error_codes.html#grouping

**Comment by zanieb** [comment]
Running `pydocstyle` on your example doesn't raise a D417 violation — which is what I was asking. I don't think they support reST docstrings at all.

**Comment by joelberkeley** [comment]
I recall using a linter that does check for this, but I'd have to dig to find which one. May have been pydoclint. I know ruff has custom checks. Might this be considered a feature request for one of those?

**Comment by joelberkeley** [comment]
this looks relevant https://github.com/astral-sh/ruff/issues/458

**Comment by joelberkeley** [comment]
indeed, i think this is a dupe of that because darglint supports sphinx rest style
